angular-client/src/components/carousel/carousel.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { Run } from 'src/utils/types.utils';
+
+// the data for the dialog, basically just all the runs
+export interface DialogData {
+  runs: Run[];
+}
+
+@Component({
+  selector: 'carousel',
+  templateUrl: 'carousel.component.html',
+  styleUrls: ['carousel.component.css']
+})
+export class Carousel {
+  runs: Run[];
+
+  responsiveOptions: any[] | undefined;
+
+  constructor(
+    public dialogRef: MatDialogRef<Carousel>,
+    @Inject(MAT_DIALOG_DATA) public data: DialogData,
+    public router: Router
+  ) {
+    this.runs = data.runs;
+  }
+
+  onNoClick(): void {
+    this.dialogRef.close();
+  }
+
+  datePipe = (time: string) => {
```

Comment:
is there a reason you need to redeclare this here and write  it differently instead of using the one in pipe.utils?

Reply:
its a different kind of format